supabase/functions/generate-json-example/index.ts

```diff
@@ -15,41 +17,71 @@ serve(async (req) => {
     return new Response('ok', { headers: corsHeaders })
   }
 
-  const { payload }: Request = await req.json();
-  const apiKey = Deno.env.get("CHATGPT_API_KEY");
-  if (!apiKey) {
-    return new Response(JSON.stringify({ message: "invalid api key" }));
-  }
-
-  const api = new ChatGPTAPI({
-    apiKey: apiKey,
-  });
-
-  let res;
   try {
-    res = await api.sendMessage(`
-      Generate an example JSON from this type,
-      do not add any other text beside the JSON example\n${payload}`);
-  } catch (e) {
-    console.error("unable to call gpt api", e);
-    return new Response(JSON.stringify({ message: "unable to call gpt api" }));
-  }
+    const redis = new Redis({
+      url: Deno.env.get("UPSTASH_REDIS_REST_URL")!,
+      token: Deno.env.get("UPSTASH_REDIS_REST_TOKEN")!,
+    });
 
-  // Find all things enclosed in curly bratches, e.g {}
-  // Hence, we can clean out all unecessary things from GPT
-  const jsonRegex = /\{[^{}]*\}/g;
-  const jsonMatches = res?.text?.match(jsonRegex);
-  if (!jsonMatches) {
-    return new Response(
-      JSON.stringify({ message: "unable to generate json example" })
-    );
-  }
+    // Create a new ratelimiter, that allows 10 requests per 10 seconds
+    const ratelimit = new Ratelimit({
+      redis,
+      limiter: Ratelimit.slidingWindow(50, "1 d"),
```

Comment:
We should put the limit in an env variable, otherwise we need to do code change everytime